### ISSUE TITLE
tf.keras.metrics.get return string intead of exception

### ISSUE THREAD

**Issue by venergiac-bh** [tensorflow/tensorflow#62345]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

binary

### TensorFlow version

2.14.0

### Custom code

No

### OS platform and distribution

Windows

### Python version

3.10

### Current behavior?

Problem on metrics `keras.metrics.get`; it should raise an exception if metric doesn't exist: InvalidIdentifier

### Standalone code to reproduce the issue

```shell
Problem on metrics

         import tensorflow as tf
         tf.keras.metrics.get('abc')

it returns "abc", but as per documentation it should raise an exception ValueError
```

**Comment by Venkat6871** [comment]
Hi **@venergiac-bh** 
I replicated this issue with Tf 2.14 ,2.13 and tf-nightly too. I attached a [gist](https://colab.research.google.com/gist/Venkat6871/0ebdf8b84d6ab18e13df7188fc821346/62345_2-14-v.ipynb) here for reference.

Thank you!

**Comment by SuryanarayanaY** [comment]
Hi @venergiac-bh ,

I have tested it with keras-nightly(3.0.0.dev2023110803) and it is raising intended exception. Please refer attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/36958e240b0ecb9583f86d19df361a73/62345.ipynb#scrollTo=41bcOz8UCedO).
